src/game/models/asteroid.rs

```diff
@@ -38,39 +57,72 @@ fn generate_circle(radius: f64) -> CircularPolygon {
     circle
 }
 
+/// This function takes in a shape (`CircularPolygon`) and mutates its vertices.
+/// There are few steps in this process:
+/// * add a (scaled) random amount to each dimension of each vertex
+/// * calculate the average location of the vertices. As a circle their average
+///   was [0,0], but the mutating changes this slightly
+/// * subtract the average position of the vertices from each of them. This
+///   ensures that the shape is roughly centered around 0. We aren't actually
+///   doing a real center-of-mass calculation, but this looks pretty good.
 fn randomize_shape(mut shape: CircularPolygon, max: f64) -> CircularPolygon {
-    let mut average_x = 0.0;
-    let mut average_y = 0.0;
+    let mut average: Vector = Default::default();
     for mut vertex in &mut shape {
-        vertex[0] += rand::random::<f64>() * max;
-        vertex[1] += rand::random::<f64>() * max;
-        average_x += vertex[0];
-        average_y += vertex[1];
+
+        // Here we create a pair of random values and add them to a vertex.
+        let rand_vect = Vector::new_rand(0.0, 0.0, max, max);
+        vertex[0] += rand_vect.x;
```

Comment:
If you were to allow that, then that would negate the point of not allowing implementations of traits you don't own on types you don't own. I think they thought doing that kind of thing lowered the bar for underhanded rust code because imported code could overload the meaning of basic operations on standard library types.